pkg/doc/jose/encrypter_test.go

```diff
@@ -0,0 +1,109 @@
+
+	"github.com/hyperledger/aries-framework-go/pkg/crypto/tinkcrypto/primitive/composite/ecdhes"
+	ecdhessubtle "github.com/hyperledger/aries-framework-go/pkg/crypto/tinkcrypto/primitive/composite/ecdhes/subtle"
+)
+
+func TestJWEEncrypt(t *testing.T) {
+	_, err := NewJWEEncrypt(nil)
+	require.Error(t, err, "NewJWEEncrypt should fail with empty recipientPubKeys")
+
+	recECKeys, _ := createRecipients(t, 20)
+
+	jweEncrypter, err := NewJWEEncrypt(recECKeys)
+	require.NoError(t, err, "NewJWEEncrypt should not fail with non empty recipientPubKeys")
+
+	jwe, err := jweEncrypter.Encrypt([]byte("some msg"), []byte("aad value"))
+	require.NoError(t, err)
+	require.Equal(t, len(recECKeys), len(jwe.Recipients))
+
+	serializedJWE, err := jwe.Serialize(json.Marshal) // must use Go Jose's json.Marshal fct (not encoding.json.Marshal)
+	require.NoError(t, err)
+	require.NotEmpty(t, serializedJWE)
+
+	t.Logf("serialized JWE is: %s", serializedJWE)
+
+	// try to deserialize with Go Jose (can't decrypt in Go Jose since private key is protected by Tink)
+	joseJWE, err := jose.ParseEncrypted(serializedJWE)
+	require.NoError(t, err)
+	require.NotEmpty(t, joseJWE)
+
+	_, err = jweEncrypter.Decrypt([]byte{})
+	require.Error(t, err, "TODO implement Decrypt()")
+}
+
+// createRecipients and return their public key and keyset.Handle
+func createRecipients(t *testing.T, numberOfRecipients int) ([]ecdhessubtle.ECPublicKey, []*keyset.Handle) {
+	t.Helper()
+
+	var (
+		r   []ecdhessubtle.ECPublicKey
+		rKH []*keyset.Handle
+	)
+
+	for i := 0; i < numberOfRecipients; i++ {
+		mrKey, kh := createAndMarshalRecipient(t)
+		ecPubKey := new(ecdhessubtle.ECPublicKey)
+		err := json.Unmarshal(mrKey, ecPubKey)
+		require.NoError(t, err)
+
+		r = append(r, *ecPubKey)
+		rKH = append(rKH, kh)
+	}
+
+	return r, rKH
+}
+
+// createAndMarshalRecipient creates a new recipient keyset.Handle, extracts public key, marshals it and returns
+// both marshaled public key and original recipient keyset.Handle
+func createAndMarshalRecipient(t *testing.T) ([]byte, *keyset.Handle) {
+	t.Helper()
+
+	kh, err := keyset.NewHandle(ecdhes.ECDHES256KWAES256GCMKeyTemplate())
+	require.NoError(t, err)
+
+	pubKH, err := kh.Public()
+	require.NoError(t, err)
+
+	buf := new(bytes.Buffer)
+	pubKeyWriter := ecdhes.NewWriter(buf)
+	require.NotEmpty(t, pubKeyWriter)
+
+	err = pubKH.WriteWithNoSecrets(pubKeyWriter)
+	require.NoError(t, err)
+
+	return buf.Bytes(), kh
+}
+
+func TestFailConvertRecKeyToMarshalledJWK(t *testing.T) {
+	recKey := &ecdhessubtle.RecipientWrappedKey{
+		EPK: ecdhessubtle.ECPublicKey{
+			Curve: "badCurveName",
+		},
+	}
+
+	_, err := convertRecKeyToMarshalledJWK(recKey)
+	require.Error(t, err)
+}
+
+func TestEmptyComputeAuthData(t *testing.T) {
+	protecteHeaders := new(map[string]interface{})
+	aad := []byte("")
+	_, err := computeAuthData(*protecteHeaders, aad)
+	require.NoError(t, err, "computeAuthData with empty protectedHeaders and empty aad should not fail")
```

Comment:
Adding these specific test comments in case of failure is a good idea... I should start doing that.